jax/_src/ad_util.py

```diff
@@ -75,8 +75,11 @@ def __repr__(self):
   @staticmethod
   def from_value(val):
     return Zero(raise_to_shaped(get_aval(val)))
+  def to_concrete(self):
+    return zeros_like_aval(self.aval)
 
 register_pytree_node(Zero, lambda z: ((), z.aval), lambda aval, _: Zero(aval))
+core.pytype_aval_mappings[Zero] = lambda z: z.to_concrete()
```

Comment:
This does not look right.  `pytype_aval_mappings` should contain functions that return `core.AbstractValue`. 

I think it would be helpful to come up with a JAX test case that demonstrates this. The test case in #7340 is too complicated.

Reply:
> This does not look right. pytype_aval_mappings should contain functions that return core.AbstractValue

Oh, I see.  How do we do that from an `aval`?  Do we just return `z.aval`?

> I think it would be helpful to come up with a JAX test case that demonstrates this.

I totally agree.  The issue is that it's incredibly time consuming for me to create MWEs since I have to basically strip out chunks of the giant program until I have one.  Is it possible to look at the tracer types and know how to produce such tracers with a minimal program?